slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -171,6 +171,11 @@ class SLAMServiceImpl final : public SLAMService::Service {
     size_t current_file_offline = 0;
     std::string current_file_online;
 
+    std::atomic<bool> optimizing{false};
+    std::string jpeg_img = "";
```

Comment:
`jpeg_img` and `buffer` need to be mutex-protected, so it is safe to access them concurrently.

Reply:
I thought they didn't need to be protected since they're only read and written within `GetMap`. My assumption was that subsequent calls to `GetMap` would be executed sequentially rather than concurrently. Was I mistaken in my assumption?

Reply:
If there are multiple app instances view the same robot, there could be concurrent calls to `GetMap`.